### ISSUE TITLE
`nan` for q values in lineage driver test

### ISSUE THREAD

**Issue by WeilerP** [theislab/cellrank#819]
## Description

When running

```
g = GPCCA(kernel)
# ...
g.compute_lineage_drivers(return_drivers=False)
```

all q values are `nan`. Is there a reasonable explanation for this, @michalk8?

**Comment by WeilerP** [comment]
Just checked: This is also the case in the real time kernel tutorial.

**Comment by Marius1311** [comment]
I've also observed this before.

**Comment by michalk8** [comment]
> Just checked: This is also the case in the real time kernel tutorial.

This can be because some gene/lineage has a standard deviation of zero, which results in division by zero, causing the correlation to be `NaN`, which are then later using for `pval/qval` calculation. Are all p-values NaN as well?

**Comment by WeilerP** [comment]
> This can be because some gene/lineage has a standard deviation of zero, which results in division by zero, causing the correlation to be NaN, which are then later using for pval/qval calculation. Are all p-values NaN as well?

No, the p values are not `nan`.

**Comment by michalk8** [comment]
Strange, this should only happen when the p-values are NaN; running `adata.varm['terminal_lineage_drivers']['Neural_1_pval'].isnull().sum()` in the WOT tutorial,
I get `36` NaN values. Looking at those genes, they all have 0 expression.

**Comment by michalk8** [comment]
Since the multiple tests correction can't handle NaN p-values, maybe we just subset non-NaN p-values for the q-value computation.

**Comment by Marius1311** [comment]
Ok, and what about the case in which the p-values are not NaN? In any case, we should issue a warning for zero-expressed genes (they should have been filtered out before by the user with any standard scanpy filtering anyways).

**Comment by michalk8** [comment]
> Ok, and what about the case in which the p-values are not NaN? In any case, we should issue a warning for zero-expressed genes (they should have been filtered out before by the user with any standard scanpy filtering anyways).

As I've suggested, for q-value calculation, let's subset only non-NaN p-values, which occurr when the gene expression for the gene is constant. We can also add the warning that some genes have no variability.

**Comment by WeilerP** [comment]
@michalk8, the q-values are also NaN when the p-values are not NaN.

**Comment by michalk8** [comment]
> @michalk8, the q-values are also NaN when the p-values are not NaN.

If any p-value is NaN, then all q-values are NaN (that's what `statsmodels` does).

**Comment by Marius1311** [comment]
I agree, let's go for your suggestion then @michalk8

**Comment by michalk8** [comment]
closed via #835
